### PR TITLE
PUB-1636 Download reference data as CSV

### PR DESCRIPTION
**Pull request by joshblackmoor**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1636

### Change description ###

Added new endpoint to get the current reference data as a CSV (same format that it was uploaded in).

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by KianKwa — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationApiTest.java [hunk 1]
```diff
@@ -784,4 +797,32 @@ void testDeleteLocationNotAuthorised() throws Exception {
             .andExpect(status().isUnauthorized())
             .andReturn();
     }
+
+    @Test
+    @WithMockUser(username = USERNAME, authorities = {VALID_ROLE})
+    void testdDownloadLocations() throws Exception {
```
Comment (joshblackmoor) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationApiTest.java:803]: Can we remove the extra d after test :)

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationApiTest.java [hunk 2]
```diff
@@ -784,4 +797,32 @@ void testDeleteLocationNotAuthorised() throws Exception {
...
+        MvcResult mvcResult = mockMvc.perform(
+                get(DOWNLOAD_LOCATIONS_ENDPOINT))
+            .andExpect(status().isOk())
+            .andReturn();
+        String returnedLocations = mvcResult.getResponse().getContentAsString();
+
+        for (Location createdLocation:createdLocations) {
+            assertTrue(
+                returnedLocations.contains(createdLocation.getName()),
+                "Location names should mach"
```
Comment (furnivall) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationApiTest.java:815]: *match

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationApiTest.java (hunk 1 above)
Comment (ashwini-mv) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationApiTest.java]: Thanks, pushed the change :)

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationApiTest.java (hunk 2 above)
Comment (ashwini-mv) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationApiTest.java]: Thanks, pushed the correction :)

**Review by furnivall — APPROVED** [review]

**Pull request merged**